common/task_executor/src/lib.rs

```diff
@@ -179,6 +179,46 @@ impl TaskExecutor {
         }
     }
 
+    /// Run a task to completion on the runtime and return the result.
+    /// The task is wrapped in an exit future and is shutdown on receiving the exit signal.
+    ///
+    /// Returns `None` if the task is shutdown before completion or if the runtime has been dropped.
+    ///
+    /// Note: This method **will panic** if called from an async context.
+    pub fn block_on<V>(&self, task: impl Future<Output = V>, name: &'static str) -> Option<V> {
```

Comment:
Not completely sure if we want to shutdown the program if this future panics.

Reply:
Hmm, I'd lean towards *not* shutting it down. I can't imagine that shutting down would help (like it does for discv5) and it carries the risk of a small, partial failure turning into a full blown liveness issue.